testsuite/model/src/test/java/org/keycloak/testsuite/model/transaction/StorageTransactionTest.java

```diff
@@ -146,6 +152,10 @@ public void testLockObjectForModification() {
         tx3.begin();
         tx3.runStep(session -> LockObjectsForModification.lockRealmsForModification(session, () -> session.realms().getRealm(realmId)));
         tx3.commit();
+
+        if (originalTimeoutValue == null) System.clearProperty(LOCK_TIMEOUT_SYSTEM_PROPERTY);
+        else System.setProperty(LOCK_TIMEOUT_SYSTEM_PROPERTY, originalTimeoutValue);
+        reinitializeKeycloakSessionFactory();
```

Comment:
When the test fails, the system property might not be cleaned up. 

Therefore wrap it in a try/finally, so it is cleaned up in all cases. 

In addition: the if/else should be wrapped with curly braces as best practice.